request.go

```diff
@@ -1,15 +1,23 @@
 package sarama
 
-type requestEncoder interface {
+import (
+	"fmt"
+)
+
+type requestBody interface {
 	encoder
+	decoder
 	key() int16
 	version() int16
 }
 
 type request struct {
+	length        int32
+	key           int16
+	version       int16
```

Comment:
key and version do not need to be members of this struct, they are already implicit in the body